src/GafferScene/Transform.cpp

```diff
@@ -98,20 +113,67 @@ bool Transform::processesTransform() const
 
 void Transform::hashProcessedTransform( const ScenePath &path, const Gaffer::Context *context, IECore::MurmurHash &h ) const
 {
-	spacePlug()->hash( h );
+	const Space space = static_cast<Space>( spacePlug()->getValue() );
+	const Mode mode = static_cast<Mode>( modePlug()->getValue() );
+	h.append( space );
+	h.append( mode );
 	transformPlug()->hash( h );
+
+	if( space == World )
+	{
+		ScenePath parentPath = path;
+		parentPath.pop_back();
+		h.append( inPlug()->fullTransformHash( parentPath ) );
+	}
+	else if( space == Parent && mode == Absolute )
+	{
+		ScenePath parentPath = path;
+		parentPath.pop_back();
+		h.append( inPlug()->transformHash( parentPath ) );
+	}
 }
 
 Imath::M44f Transform::computeProcessedTransform( const ScenePath &path, const Gaffer::Context *context, const Imath::M44f &inputTransform ) const
 {
-	Space space = static_cast<Space>( spacePlug()->getValue() );
-	Imath::M44f matrix = transformPlug()->matrix();
+	const Space space = static_cast<Space>( spacePlug()->getValue() );
+	const Mode mode = static_cast<Mode>( modePlug()->getValue() );
+	const Imath::M44f matrix = transformPlug()->matrix();
 	if( space == World )
 	{
-		return inputTransform * matrix;
+		ScenePath parentPath = path;
+		parentPath.pop_back();
+		const Imath::M44f parentMatrix = inPlug()->fullTransform( parentPath );
```

Comment:
To resolve the Absolute World across ... case, I changed this line to:

const Imath::M44f parentMatrix = outPlug()->fullTransform( parentPath );